cpp/modules/luma.gl/garrow/src/util/arrow-utils.cc

```diff
@@ -123,22 +122,29 @@ auto vertexFormatFromArrowType(const std::shared_ptr<arrow::DataType>& type) ->
   return std::nullopt;
 }
 
-auto transformTable(const std::shared_ptr<arrow::Table>& table, const std::vector<AttributeDescriptor>& descriptors,
+auto transformTable(const std::shared_ptr<arrow::Table>& table, const std::vector<ColumnBuilder>& builders,
```

Comment:
Thoughts (for upcoming discussions):

I am still trying to understand the builders, but it looks like this is no longer a simple function that copies an existing table's columns to the GPU.  This is now a function that maps attributes and does additional work, before copying to GPU.

- For efficiency's stake we might certainly want a function that does both,
- but a primary thing we want to offer is a table API with simple primitive operations that are composable.

Additionally on naming:
- `transformTable` is way to generic as my hope is that this will be a growing family of functions. Since it now does multiple things it is a little harder to name. 
- I do feel it would be very helpful for users to talk about the GPU side table as `GPUTable`. It would be a partial departure from the `garrow` namespace idea. 
- I think about how to explain this API to users, and how to name these transform functions. `copyToGPUTable`, `makeGPUTable`, ... ?

Reply:
> I am still trying to understand the builders, but it looks like this is no longer a simple function that copies an existing table's columns to the GPU. This is now a function that maps attributes and does additional work, before copying to GPU.

This has been the case all along, and that's something I wanted to discuss. As you noted, this could be one of the, if not the biggest potential bottlenecks that should be as performant as possible. In order to avoid the intermediate step of creating an intermediate `arrow::Table`, the accessor functions would need to create GPU arrays instead. I was under impression that's something we wanted to abstract away from the user, but it might be a big tradeoff for us to make.

The idea behind `ColumnBuilder` is that whoever is going to be transforming the input `arrow::Table` should provide two things that allow us to build a new set of columns:
- A `Field` that describes the output column
- A mapping function that transforms each columns data